In [3]:
# se importan las librerias

import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [4]:
# se realiza el cargue de los datos en un dataframe

data = pd.read_csv('heart_failure_clear.csv')

In [22]:
# se definen las variables

Z= data
Z= (Z.drop(['DEATH_EVENT', 'age', 'age_group'], axis = 1)) # se eliminan las columnas evento de muerte, edad, grupo etario 
X= Z[['anaemia','diabetes','ejection_fraction','high_blood_pressure','platelets','serum_creatinine','sex','smoking']]
y = data['age']

d = np.corrcoef(X)
print(len(X))
print(len(y))

225
225


In [24]:
# se crea una instancia del modelo de regresión

regression = LinearRegression()

# Ajustar el modelo a los datos
regression.fit(X, y)

LinearRegression()

In [34]:
# Se realiza predicción 
y_pred = regression.predict(X)
print("Predicciones:", y_pred) # predicción de edades
print("Edad:", y)              # edades reales

Predicciones: [68.16993886 60.66494319 67.17360466 76.89452913 60.02339549 67.40260597
 58.60431742 61.03213085 63.04951607 58.04848439 57.59808914 53.17861491
 60.88968517 63.55792249 66.27084189 60.86611216 55.37125659 68.12284271
 69.44206607 63.98595388 64.58024186 59.2801649  69.11554403 59.80476776
 58.18478942 62.96777573 60.20278685 62.08327141 68.2059376  63.86117595
 65.32635955 62.39724196 63.7351457  66.38020152 55.2126652  61.99716214
 61.22185999 61.51273764 56.17227097 58.85025087 61.28987691 58.95855806
 54.53016082 58.69747744 57.29927475 58.18338379 54.03867151 58.12580121
 64.47543501 57.55044906 54.44840925 61.89857946 59.32044335 56.9637776
 59.78513671 65.2512699  62.00644321 74.83710403 58.58371074 53.03739519
 63.14296505 61.43485961 57.11964878 62.848539   56.11825735 56.67826808
 62.99316626 65.67249692 55.30392627 66.39447736 62.66370971 63.64487831
 60.55271354 60.13294634 57.17906877 63.87822929 59.41106675 61.80462218
 58.47097629 62.96717532 60.58966439 6

In [35]:
# evaluación de resultados

mse = mean_squared_error(y, y_pred)
print("Error cuadrático medio (MSE):", mse)

r_squared = regression.score(X, y)
print("Coeficiente de determinación (R²):", r_squared)

Error cuadrático medio (MSE): 122.95794765083113
Coeficiente de determinación (R²): 0.13067323152429455


# Analisis de resultados

Al comparar los resultados de la edades obtenidas mediante la regresión se evidencia que estas son difierentes de las edades reales. Por otra parte el MSE nos indica cuanto estan desviados de forma cuadratica los datos predichos respecto a los datos reales y el R² nos indica que tanto explican las variables independientes a la variable dependiente para este caso el valor de R² es 0.13 lo cual no es bueno dado que un valor aceptable para R² iniciaria desde 0.8 , por lo ane